test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -790,7 +790,7 @@ public static void assertToXContentEquivalent(BytesReference expected, BytesRefe
             Map<String, Object> actualMap = actualParser.map();
             try (XContentParser expectedParser = xContentType.xContent().createParser(NamedXContentRegistry.EMPTY, expected)) {
                 Map<String, Object> expectedMap = expectedParser.map();
-                assertMapEquals(expectedMap, actualMap);
+                assertMapEquals(expectedMap, actualMap, "");
```

Comment:
As mentioned in another comment, I added these path arguments to be able to better debug errors in map-comparisons when we compare xContent. This empty path is used as initial value, assertMapEquals() will be called recursively later.